pkg/harvester/dialog/HarvesterAddHotplugNic.vue

```diff
@@ -0,0 +1,201 @@
+<script>
+import { exceptionToErrorsArray } from '@shell/utils/error';
+import { mapGetters } from 'vuex';
+import { NETWORK_ATTACHMENT } from '@shell/config/types';
+import { Card } from '@components/Card';
+import { Banner } from '@components/Banner';
+import AsyncButton from '@shell/components/AsyncButton';
+import { LabeledInput } from '@components/Form/LabeledInput';
+import LabeledSelect from '@shell/components/form/LabeledSelect';
+import { HCI as HCI_ANNOTATIONS } from '@pkg/harvester/config/labels-annotations';
+import { NETWORK_TYPE } from '../config/types';
+
+export default {
+  name: 'AddHotplugNic',
+
+  emits: ['close'],
+
+  components: {
+    AsyncButton,
+    Card,
+    LabeledInput,
+    LabeledSelect,
+    Banner
+  },
+
+  props: {
+    resources: {
+      type:     Array,
+      required: true
+    }
+  },
+
+  async fetch() {
+    try {
+      this.allVMNetworks = await this.$store.dispatch('harvester/findAll', { type: NETWORK_ATTACHMENT });
+    } catch (err) {
+      this.errors = exceptionToErrorsArray(err);
+      this.allVMNetworks = [];
+    }
+  },
+
+  data() {
+    return {
+      interfaceName: '',
+      networkName:   '',
+      macAddress:    '',
+      allVMNetworks: [],
+      errors:        [],
+    };
+  },
+
+  computed: {
+    ...mapGetters({ t: 'i18n/t' }),
+
+    actionResource() {
+      return this.resources?.[0];
+    },
+
+    isFormValid() {
+      return this.interfaceName !== '' && this.networkName !== '';
+    },
+
+    vmNetworksOption() {
+      return this.allVMNetworks
+        .filter((network) => {
+          const labels = network.metadata?.labels || {};
+          const type = labels[HCI_ANNOTATIONS.NETWORK_TYPE];
+
+          return [NETWORK_TYPE.L2VLAN, NETWORK_TYPE.UNTAGGED, NETWORK_TYPE.L2TRUNK_VLAN].includes(type);
+        })
+        .map((network) => ({
+          label: network.metadata?.name || '',
```

Comment:
In case there are identical names of NADs between namespaces, could we use `network.id` as the label such that it can display distinguishable names like `ns1/nad1` and `ns2/nad1`?

Additionally, it seems useful and good to have notReady suffix like [here](https://github.com/harvester/harvester-ui-extension/blob/6f90cae482d18842fcf3afc6651343a3053fa7ac/pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachineNetwork/index.vue#L54) if it's not too hard to add.

Reply:
Updated in commit https://github.com/harvester/harvester-ui-extension/pull/582/commits/b3b10ee5990c0ed01c3c2b35a9e4978c20659883.
<img width="1920" height="968" alt="Screenshot 2025-11-05 at 11 13 22 AM (2)" src="https://github.com/user-attachments/assets/c9c4e494-38ba-4fd8-af9a-3d1f8137a501" />